src/MediaUploader.php

```diff
@@ -694,6 +700,33 @@ private function deleteExistingMedia(Media $model)
             ->where('extension', $model->extension)
             ->delete();
 
+        $this->deleteExistingFile($model);
+    }
+
+    /**
+     * Find the media that previously existed at a destination.
+     * @param  \Plank\Mediable\Media  $model
+     * @return \Plank\Mediable\Media  $model
+     */
+    private function findExistingMediaRecord(Media $model)
+    {
+        if ($model->exists) {
+            return $model;
+        }
+        return Media::where('disk', $model->disk)
+            ->where('directory', $model->directory)
+            ->where('filename', $model->filename)
+            ->where('extension', $model->extension)
+            ->first();
```

Comment:
you could still condense this part into a single query scope (basically mirror `scopeForPathOnDisk()` with filename and extension instead of basename). You can still leave this method if you want to check for existence first.